### PR TITLE
Allows @AutoConfigureWireMock to respect user-specified properties 

### PR DESCRIPTION
**Pull request by bono007**
Allows @AutoConfigureWireMock to respect user-specified properties in application.yml

Fixes gh-1555

### CODE REVIEW

File: spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/AutoConfigureWireMock.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 @Import(WireMockConfiguration.class)
-@PropertyMapping("wiremock.server")
+@PropertyMapping(value = "wiremock.server", skip = SkipPropertyMapping.ON_DEFAULT_VALUE)
```
Comment (bono007) [inline spring-cloud-contract-wiremock/src/main/java/org/springframework/cloud/contract/wiremock/AutoConfigureWireMock.java:44]: This is the fix, by comparing/contrasting w/ our tried-true friend [AutoConfigureMockMvc](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-test-autoconfigure/src/main/java/org/springframework/boot/test/autoconfigure/web/servlet/AutoConfigureMockMvc.java#L63) I was able to see the magic that we needed to apply here.

File: spring-cloud-contract-wiremock/src/test/java/org/springframework/cloud/contract/wiremock/AutoConfigureWireMockConfigurationInYamlTests.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = WiremockTestsApplication.class,
+		properties = "app.baseUrl=http://localhost:${wiremock.server.port}",
+		webEnvironment = WebEnvironment.NONE)
+@AutoConfigureWireMock
+@ActiveProfiles("customfoo")
+public class AutoConfigureWireMockConfigurationInYamlTests {
```
Comment (bono007) [inline spring-cloud-contract-wiremock/src/test/java/org/springframework/cloud/contract/wiremock/AutoConfigureWireMockConfigurationInYamlTests.java:36]: I went ahead and added a test for this case as the other tests did cover the functionality in some form/fashion but they also specified either the `files` or `stubs` to point to content that ultimately returned the same `Hello world` message so we could not really tell if it was the defaults being used or the custom ones. For that reason I also created an explicit resources directory for the `files` and `stubs`.

File: spring-cloud-contract-wiremock/src/test/resources/application-customfoo.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+wiremock:
+  server:
+    port: 0
+    stubs:
+      - "file:src/test/resources/custom-from-yml/mappings"
```
Comment (bono007) [inline spring-cloud-contract-wiremock/src/test/resources/application-customfoo.yml:5]: Again, pointing to these custom dirs assures us that the user overrides here in the yaml are being used.

File: spring-cloud-contract-wiremock/src/test/resources/custom-from-yml/__files/hello-custom.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "message": "Hello Custom One"
```
Comment (bono007) [inline spring-cloud-contract-wiremock/src/test/resources/custom-from-yml/__files/hello-custom.json:2]: Again, a custom message tailored specific for this case.

**Comment by marcingrzejszczak** [comment]
Great work @bono007, as usual!

**Pull request merged**